src/main/java/org/craftercms/deployer/impl/processors/AbstractSearchIndexingProcessor.java

```diff
@@ -208,8 +210,9 @@ protected ChangeSet getFilteredChangeSet(ChangeSet changeSet) {
             logger.info("Ensuring that index {} exists", indexId);
             doCreateIndexIfMissing();
         }
+        boolean isReprocessAll = BooleanUtils.toBoolean(getDeploymentParam(REPROCESS_ALL_FILES_PARAM_NAME));
```

Comment:
I think the safest approach is to change the signature of the `getFilteredChangeSet`  method to pass the `Deployment` instance

Reply:
That would break any processor overriding that method though

Reply:
Another solution that might be to use a thread local: https://docs.oracle.com/en/java/javase/21/core/thread-local-variables.html#GUID-2CEB9041-3DF7-43DA-868F-E0596F4B63FD

Reply:
I will take a look at that, we might already have some thread local variables that can help